### PR TITLE
[GEOS-8010] dbtype and filetype store arguments should not be visible

### PR DESCRIPTION
**Pull request by tbarsballe**
See https://osgeo-org.atlassian.net/browse/GEOS-8010

Depends upon https://github.com/geotools/geotools/pull/1659. Travis CI will fail until that is merged.

Supersedes https://github.com/geoserver/geoserver/pull/2449; only one of these PRs should be merged.

### CODE REVIEW

File: src/web/core/src/test/java/org/geoserver/web/data/store/DataAccessNewPageDatabaseTest.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
+package org.geoserver.web.data.store;
```
Comment (jodygarnett) [inline src/web/core/src/test/java/org/geoserver/web/data/store/DataAccessNewPageDatabaseTest.java]: missing header

File: src/web/core/src/main/java/org/geoserver/web/data/store/ParamInfo.java [hunk 2]
```diff
@@ -105,7 +108,11 @@ public String getTitle() {
     public boolean isPassword() {
         return password;
     }
-    
+
+    public String getLevel() {
```
Comment (jodygarnett) [inline src/web/core/src/main/java/org/geoserver/web/data/store/ParamInfo.java]: thanks, working around ParamInfo is frustrating, do we need javadocs for these public methods (say to define the levels).

Comment (tbarsballe) [inline src/web/core/src/main/java/org/geoserver/web/data/store/ParamInfo.java]: ParamInfo more or less directly mirrors Param / Parameter, and is only used by the UI code.
Javadocs defining the level are accessible from the Parameter javadocs for the equivalent methods.
Ultimately, the information already exists in a more useful location, although we could maybe reference those classes?

File: src/web/core/src/test/java/org/geoserver/web/data/store/DataAccessNewPageDatabaseTest.java (hunk 1 above)
Comment (tbarsballe) [inline src/web/core/src/test/java/org/geoserver/web/data/store/DataAccessNewPageDatabaseTest.java]: Indeed. Fixed.

**Comment by tbarsballe** [comment]
@jodygarnett - Are you happy with the changes here? Do you want to mark this as approved.

**Pull request merged**